// labeled-switch-material-ui/lib/Layout.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var styles = {
    center: {
        alignItems: 'center',
        display: 'inline-flex',
        margin: '0 5px'
    },
    container: {
        alignItems: 'center',
        display: 'inline-flex'
    },
    left: {
        alignItems: 'center',
        display: 'inline-flex'
    },
    right: {
        alignItems: 'center',
        display: 'inline-flex'
    }
};
var getStyleContainer = function (style) {
    if (style === undefined) {
        return styles.container;
    }
    return __assign(__assign({}, styles.container), style);
};
var Layout = React.memo(function (props) { return (React.createElement("div", { style: getStyleContainer(props.style) },
    React.createElement("div", { style: styles.left }, props.left),
    React.createElement("div", { style: styles.center }, props.center),
    React.createElement("div", { style: styles.right }, props.right))); });
exports.default = Layout;
